src/chunk_constraint.c

```diff
@@ -444,6 +444,63 @@ typedef struct ChunkConstraintScanData
 	DimensionSlice *slice;
 } ChunkConstraintScanData;
 
+/*
+ * Scan for all chunk constraints that match the given slice ID. The chunk
+ * constraints are not saved, only the matching chunk id is returned.
+ */
+int
+ts_chunk_constraint_scan_by_dimension_slice_chunk_id(const DimensionSlice *slice, ChunkScanCtx *ctx,
+													 MemoryContext mctx)
+{
+	int result = 0;
+
+	ScanIterator iterator = ts_scan_iterator_create(CHUNK_CONSTRAINT, AccessShareLock, mctx);
+	ts_chunk_constraint_scan_iterator_set_slice_id(&iterator, slice->fd.id);
+
+	ts_scanner_foreach(&iterator)
+	{
+		ChunkScanEntry *entry;
+		bool found;
+		TupleInfo *ti = ts_scan_iterator_tuple_info(&iterator);
+		Datum datum = slot_getattr(ti->slot, Anum_chunk_constraint_chunk_id, &found);
+		int32 chunk_id = DatumGetInt32(datum);
+
+		if (slot_attisnull(ts_scan_iterator_slot(&iterator),
+						   Anum_chunk_constraint_dimension_slice_id))
+			continue;
```

Comment:
I suggest moving this check about `slot_getattr` above, since there is no need to get the `chunk_id` if this chunk isn't true.

Reply:
Removed this check.